source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1264,9 +1282,11 @@ implementation::
                 return
 
             if isStateChangeError(error):
-                # Mark the server Unknown
```

Comment:
In Java driver I implemented error handling for load balancer in a separate method, so I can't compare it directly to this pseudocode.  But pseudocode doesn't look right.  For example:

- There's no clearing of connection pool on network error when in LoadBalanced mode.
- The clearing of the pool when there is a shutdown error clears the whole pool and doesn't mention anything about serviceId

Reply:
I've added the conditions for clearing the pool for service id now in both blocks.